src/Pecee/Pixie/Exception.php

```diff
@@ -97,13 +96,6 @@ public static function create(\Exception $e, string $adapterName = null, QueryOb
                             return new NotNullException($e->getMessage(), $errorCode, $e->getPrevious(), $query);
                     }
                     break;
-                case Sqlite::class:
-                    // https://sqlite.org/c3ref/c_abort.html
-                    switch ($errorCode) {
-                        case \SQLITE_CONSTRAINT: // Abort due to constraint violation
-                            return new NotNullException($errorMsg, $errorCode, $e->getPrevious(), $query);
-                    }
```

Comment:
What about making `ext-sqlite` a [suggested Composer package](https://getcomposer.org/doc/04-schema.md#suggest) and in your exception handling make use of [`defined`](http://php.net/manual/en/function.defined.php) or [`constant`](http://php.net/manual/en/function.constant.php)?

```php
case Sqlite::class:
    if (!defined('SQLITE_CONSTRAINT')) {
        // something here
        return;
    }

    // https://sqlite.org/c3ref/c_abort.html
    switch ($errorCode) {
        case \SQLITE_CONSTRAINT: // Abort due to constraint violation
            return new NotNullException($errorMsg, $errorCode, $e->getPrevious(), $query);
    }
    break;
```

Reply:
Not only `ext-sqlite` but also `ext-sqlite3`.